contracts/VaultToken.vy

```diff
@@ -0,0 +1,186 @@
+# @version 0.2.16
+
+API_VERSION: constant(String[28]) = "0.5.0"
+
+vault: public(address)
+gov: public(address)
+
+event Transfer:
+    sender: indexed(address)
+    receiver: indexed(address)
+    value: uint256
+
+event Approval:
+    owner: indexed(address)
+    spender: indexed(address)
+    value: uint256
+
+decimals: public(uint256)
+balanceOf: public(HashMap[address, uint256])
+allowance: public(HashMap[address, HashMap[address, uint256]])
+totalSupply: public(uint256)
+
+# `nonces` track `permit` approvals with signature.
+nonces: public(HashMap[address, uint256])
+DOMAIN_SEPARATOR: public(bytes32)
+DOMAIN_TYPE_HASH: constant(bytes32) = keccak256('EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)')
+PERMIT_TYPE_HASH: constant(bytes32) = keccak256("Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)")
+
+interface IVaultAPI:
+    def token() -> address: view
+    def vault_token() -> address: view
+    def decimals() -> uint256: view
+    def vaultToken() -> address: view
+
+@external
+def __init__(
+    gov: address,
+    decimals: uint256
+):
+    self.gov = gov
+    self.decimals = decimals
+    # EIP-712
+    self.DOMAIN_SEPARATOR = keccak256(
+        concat(
+            DOMAIN_TYPE_HASH,
+            keccak256(convert("Yearn Vault", Bytes[11])),
+            keccak256(convert(API_VERSION, Bytes[28])),
```

Comment:
The domain separator contains the API_VERSION, I am not sure if this should be fetched from the vault since our goal is to update vaults without updating the VaultToken contract. The downside of fetching it from the vault is that if a transaction that includes a signature is broadcated/store while we are updating the vault, the signature won't be valid anymore.

Reply:
You can skip it, I think `verifyingContract` and `chainId` fields would provide enough protection.

Reply:
Are you sure, if we remove that part would it be possible to "replay" the signatures on a newer vault?